### PR TITLE
add `strip_whitespace` option for `ConstrainedStr` and `constr`

### PR DESCRIPTION
**Pull request by Gr1N**

### CODE REVIEW

**Comment by Gr1N** [comment]
@samuelcolvin updated, please check

File: HISTORY.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ v0.8.1 (2018-XX-XX)
 ...................
 * tweak email-validator import error message #145
 * fix parse error of parse_date() and parse_datetime() when input is 0 #144
+* **breaking change**: add ``Config.anystr_strip_whitespace`` and ``strip_whitespace`` kwarg to ``constr``, by default values is `True` #163
```
[the remark this answers is not in the document]
Comment (Gr1N) [reply HISTORY.rst]: In my experience we always need to strip any string data during validation except passwords or tokens. Also many validation libraries use strip logic enabled by default. But you are the owner and I will do what you prefer.

[the remark this answers is not in the document]
Comment (samuelcolvin) [reply HISTORY.rst]: ok, makes sense. This would require a patch release anyway, so let's just go with this.

**Comment by samuelcolvin** [comment]
Thanks a lot, I'll try and merge a couple of other PRs which are finished and deploy in the next couple of days.

**Comment by Gr1N** [comment]
And I will try to implement more things from #161 😃

**Pull request merged**